snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -274,10 +240,13 @@ def launch_instance(self) -> None:
             # and do first boot related things and if any failure occurs,
             # then clean up.
             self._setup_environment()
-        else:
-            # We always setup snapcraft after a start to bring it up to speed with
-            # what is on the host
-            self._setup_snapcraft()
+
+        # We always setup snapcraft after a start to bring it up to speed with
+        # what is on the host
+        self._setup_snapcraft()
+
+        # Ensure package cache is ready.
+        self._run(["snapcraft", "refresh"])
```

Comment:
it is a requirement to only do this on project setup to avoid slow turnaround on multiple calls to snapcraft

Reply:
But if the container is launched several days later, how are you sure that the cache is good?